.github/workflows/check-nix-long.sh

```diff
@@ -0,0 +1,11 @@
+NIX_DEBUG_COMMAND="" && [[ $ACTIONS_RUNNER_DEBUG = "true" ]] && NIX_DEBUG_COMMAND='--print-build-logs --debug --show-trace --verbose'
+set -o pipefail -o errexit
+NIXPKGS_ALLOW_BROKEN=1 nix flake check --keep-going --no-build --allow-import-from-derivation  --no-update-lock-file --fallback -L ${NIX_DEBUG_COMMAND} --impure --option sandbox relaxed --impure 2>&1 | tee "nix.check.log"  || true
+set +o pipefail +o errexit
+echo "exited with(https://github.com/NixOS/nix/issues/7464) ${$?}" 
+cat "nix.check.log" | grep --invert-match  "error: path [']/nix/store/[a-zA-Z0-9]\+-[a-zA-Z0-9\.-]\+['] is not valid" \
+| grep --invert-match  "error: cannot substitute path [']/nix/store/[a-zA-Z0-9]\+-[a-zA-Z0-9\.-]\+['] \- no write access to the Nix store" \
+| grep --invert-match '^error: some errors were encountered during the evaluation' > "filtered.nix.check.log"
+RESULT=$(cat "filtered.nix.check.log" | grep -c 'error:')
```

Comment:
```suggestion
RESULT=$(grep -c 'error:' "filtered.nix.check.log")
```

Reply:
haha :) fixed in next PR

Reply:
so it is like `buildin.any`

grep is not responsible for reading stream, just consume (F#/C#/Ocaml style). 

you may compose cat with anything, while direct grep repl is not composable.

Reply:
please comment. 

so change in next pr for zombienet. so IMHO better write piped script so these compose better.

Reply:
> grep is not responsible for reading stream, just consume (F#/C#/Ocaml style).

hmm... but `cat` also consumes a source wholly.

> while direct grep repl is not composable.

really?

```
grep foo bar.txt | grep ERROR | grep -v timeout | ...
```

Reply:
your first grep should match all lines.  other then that sure can grep 2 times.

i did not know that cat is not streaming. i thought it is. 
 
also would perfer timeout to be separate command, so can timeout regardless of grep or not. like separate timeout worki with any stream.